### PR TITLE
feat(): implement assertion conditions

### PR DESCRIPTION
**Pull request by jordantsanz**
1. Description:
Adds conditions to assertion logic. A condition dictates if an assertion should be checked or not. If the condition evaluates to false, the assertion can be ignored. 

We implement this logic at the checkAssertions level, as we want to determine if an assertion is relevant first before evaluating it. 

Use case: Groups with `isViewOnly: false` allow any group member to share content; groups with `isViewOnly: true` allow only group admins to share content. To use just one permission for sharing content, we use a condition on `entity.isViewOnly` to understand which assertion currently applies: `is-group-member` or `is-group-admin`. 

1. Instructions for testing:

1. Closes Issues: #9090 (if appropriate)

1. [X] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [X] used semantic commit messages
  
1. [X] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [X] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/permissions/_internal/checkAssertions.ts [hunk 1]
```diff
@@ -18,9 +18,29 @@ export function checkAssertions(
   // Only return a check if the policy is defined
   if (policy.assertions) {
     // iterate over the assertions, creating a check for each entry
-    checks = policy.assertions.map((assertion: IPolicyAssertion) => {
-      return checkAssertion(assertion, entity, context);
-    });
+    checks = policy.assertions.reduce(
+      (acc: IPolicyCheck[], assertion: IPolicyAssertion) => {
+        let conditionResult = true;
```
Comment (juliannemarik) [inline packages/common/src/permissions/_internal/checkAssertions.ts:23]: minor, but I might suggest naming this something like `shouldCheckAssertion` to make it a bit more clear?

**Review by juliannemarik — APPROVED** [review]
This looks good to me and I _believe_ achieves what we need, but let's check with @dbouwman to make sure he's okay with this enhancement to permission assertions 👍

**Comment by jordantsanz** [comment]
Requesting @dbouwman for review to confirm he's okay with these enhancements to assertions.

**Review by dbouwman — APPROVED** [review]
This looks like a good addition!

**Pull request merged**